docs/proposals/failover/optimize-cluster-failover.md

```diff
@@ -0,0 +1,635 @@
+
+#### Story 2
+
+As an application maintenance personnel, one should be able to customize the migration strategy for the cluster failure scenario:
+
+- For general workloads, when a hard failure occurs in the cluster (such failures will affect most of the workloads), trigger the migration of most workloads on the cluster;
+- For workloads with high resource requirements, when a partial failure occurs in the cluster (such failures will only affect a part of the workloads, causing some workloads to be unable to run or their performance to be affected), only trigger the migration of such workloads.
+
+#### Story 3
+
+As a conservative and cautious user, a large number of businesses adopt a fixed redundant deployment scheme, and I don't want any non-human-initiated migrations to occur in the event of a cluster failure. A small number of businesses support a dynamic HA deployment scheme, and I hope that when a cluster failure occurs, the affected parts of these businesses can be migrated to a new healthy cluster.
+
+## Design Details
+
+The overall process of Cluster failover is as follows:
+
+![cluster failover overview](optimize-cluster-failover-overview.png)
+
+Regarding the overall process of Cluster Failover, after re-sorting, it is divided into the following stages:
+1. Input of cluster status conditions.
+2. Matching of cluster failure strategies.
+3. System response to failures (preventing the scheduling of new workloads or triggering workload failure migration).
+
+### Overall Design Principles
+
+Below, we present the overall design principles of Cluster Failover:
+
+1. Judgment of cluster failure conditions
+- The sources for judging cluster failures are diverse. The cluster status detection that Karmada supports by default will first be reflected in `cluster.status.conditions`.
+- After the Karmada system determines that a cluster has a failure, it will add corresponding taints to the cluster object. In cases of severe failures, the migration process will be triggered.
+- Users can define "which taints should be added to the cluster object when certain conditions occur" by configuring the "taint cluster by conditions" rule.
+- For more complex scenarios of cluster failure judgment, users can directly implement a custom "cluster taint controller" to control how to add and remove taints from the cluster.
+
+2. Reconsider the types of taints and adjust them into the following categories according to the severity of cluster failures and the scope of affected workloads:
+- Impact new workloads' scheduling and running: taint{effect: NoSchedule}. New workloads will not be scheduled, and will not trigger resource migration.
+- Impact all workloads' scheduling and running: taint{effect: NoExecute}(Reserve semantics: Hard migration, requiring users to add relevant comments to remind them of the impact in the API definition). New workloads will not be scheduled, and existing workloads will be migrated.
```

Comment:
> requiring users to add relevant comments to remind them of the impact in the API definition

`users` to add the comments?

Reply:
I'm wrong here, we need to add comments to the API.